app/src/main/java/net/squanchy/tweets/view/TweetsAdapter.kt

```diff
@@ -0,0 +1,39 @@
+package net.squanchy.tweets.view
+
+import android.content.Context
+import android.support.v7.widget.RecyclerView
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+
+import java.util.ArrayList
+import java.util.Collections
+
+import net.squanchy.R
+import net.squanchy.tweets.domain.view.TweetViewModel
+
+class TweetsAdapter(private val context: Context) : RecyclerView.Adapter<TweetViewHolder>() {
+
+    private var tweets: List<TweetViewModel> = emptyList()
+    private lateinit var listener: TweetItemView.OnTweetClickedListener
+
+    val isEmpty: Boolean
```

Comment:
`val isEmpty = tweets.isEmpty`?

Reply:
Unfortunately no, if we did that the value would be assigned when the class instance is created and never changed. But using a custom get we can expose it as a property but the returned value is evaluated when we access it

Reply:
Figured that was the case, but then, we should have the same problem with `size()` below, no?

Reply:
`size` is implemented the same way in kotlin, exposed as a property but evaluated at "access time"

Reply:
:+1: